lib/ansible/plugins/connection/ssh.py

```diff
@@ -578,11 +589,20 @@ def _build_command(self, binary, subsystem, *other_args):
 
         # If we want to use password authentication, we have to set up a pipe to
         # write the password to sshpass.
-
-        if conn_password:
+        pkcs11_provider = self.get_option("pkcs11_provider")
+        if conn_password or len(pkcs11_provider) >= 1:
```

Comment:
I think we should keep with the normal style here and make the checks simpler. A string is `True` when it's not empty so this is the same.

```suggestion
        if conn_password or pkcs11_provider:
```